modules/oe_translation_cdt/src/Controller/CallbackController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Site\Settings;
+use Drupal\oe_translation_cdt\CdtLanguageMapper;
+use Drupal\oe_translation_cdt\TranslationRequestCdtInterface;
+use OpenEuropa\CdtClient\Serializer\CallbackSerializer;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Handles endpoints for CDT callbacks.
+ */
+class CallbackController extends ControllerBase {
+
+  /**
+   * Verifies if the API key matches the one from module settings.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return bool
+   *   TRUE if the API key is valid, FALSE otherwise.
+   */
+  private function isApiKeyValid(Request $request): bool {
+    $request_api_key = $request->headers->get('apikey');
+    $site_api_key = Settings::get('cdt.api_key');
```

Comment:
This will have to be added to the requirements, but I would wait with it until we discuss D8AGE-394